src/start_menu.c

```diff
@@ -224,82 +363,1069 @@ static void BuildStartMenuActions_BattlePyramid(void)
+            CreateTask(&sub_8153688, 0x5);
+            *step = 6;
+            break;
+        case 6:
+            if (!FuncIsActiveTask(&sub_8153688))
+            {
+                *step = 3;
+            }
+            break;
+        }
+    }
+}
+
+static void ShowSaveInfoWindow(void)
+{
+    struct WindowTemplate saveInfoWindow = gSaveInfoWindowTemplate;
+    u8 gender;
+    u8 color;
+    u32 xOffset;
+    u32 yOffset;
+    
+    if (!FlagGet(FLAG_SYS_POKEDEX_GET))
+    {
+        saveInfoWindow.height -= 2;
+    }
+
+    sSaveInfoWindowId = AddWindow(&saveInfoWindow);
+    NewMenuHelpers_DrawStdWindowFrame(sSaveInfoWindowId, FALSE);
+
+    gender = gSaveBlock2Ptr->playerGender;
+    color = TEXT_COLOR_RED;  // Red when female, blue when male.
+
+    if (gender == MALE)
+    { 
+        color = TEXT_COLOR_BLUE;
+    }
+
+    // Print region name
+    yOffset = 1;
+    sub_819A344(3, gStringVar4, TEXT_COLOR_GREEN);
+    PrintTextOnWindow(sSaveInfoWindowId, 1, gStringVar4, 0, yOffset, 0xFF, NULL);
+
+    // Print player name
+    yOffset = 0x11;
+    PrintTextOnWindow(sSaveInfoWindowId, 1, gText_SavingPlayer, 0, yOffset, 0xFF, NULL);
+    sub_819A344(0, gStringVar4, color);
+    xOffset = GetStringRightAlignXOffset(1, gStringVar4, 0x70);
+    PrintPlayerNameOnWindow(sSaveInfoWindowId, gStringVar4, xOffset, yOffset);
+
+    // Print badge count
+    yOffset = 0x21;
+    PrintTextOnWindow(sSaveInfoWindowId, 1, gText_SavingBadges, 0, yOffset, 0xFF, NULL);
+    sub_819A344(4, gStringVar4, color);
+    xOffset = GetStringRightAlignXOffset(1, gStringVar4, 0x70);
+    PrintTextOnWindow(sSaveInfoWindowId, 1, gStringVar4, xOffset, yOffset, 0xFF, NULL);
+
+    if (FlagGet(FLAG_SYS_POKEDEX_GET) == TRUE)
+    {
+        // Print pokedex count
+        yOffset = 0x31;
+        PrintTextOnWindow(sSaveInfoWindowId, 1, gText_SavingPokedex, 0, yOffset, 0xFF, NULL);
+        sub_819A344(1, gStringVar4, color);
+        xOffset = GetStringRightAlignXOffset(1, gStringVar4, 0x70);
+        PrintTextOnWindow(sSaveInfoWindowId, 1, gStringVar4, xOffset, yOffset, 0xFF, NULL);
+    }
+
+    // Print play time
+    yOffset += 0x10;
+    PrintTextOnWindow(sSaveInfoWindowId, 1, gText_SavingTime, 0, yOffset, 0xFF, NULL);
+    sub_819A344(2, gStringVar4, color);
+    xOffset = GetStringRightAlignXOffset(1, gStringVar4, 0x70);
+    PrintTextOnWindow(sSaveInfoWindowId, 1, gStringVar4, xOffset, yOffset, 0xFF, NULL);
+
+    CopyWindowToVram(sSaveInfoWindowId, 2);
+}
+
+static void RemoveSaveInfoWindow(void)
+{
+    sub_819746C(sSaveInfoWindowId, FALSE);
+    RemoveWindow(sSaveInfoWindowId);
+}
+
+static void sub_80A08A4(u8 taskId)
+{
+    if (!FuncIsActiveTask(sub_8153688))
+    {
+        DestroyTask(taskId);
+        EnableBothScriptContexts();
+    }
+}
+
+void sub_80A08CC(void) // Referenced in data/specials.inc and data/scripts/maps/BattleFrontier_BattleTowerLobby.inc
+{
+    u32 taskId1;
+    u32 taskId2;
+
+    taskId1 = CreateTask(&sub_8153688, 0x5);
```

Comment:
Try ditching the temporary variables, taskId temps are basically never given u32 typing, so it was likely a direct CreateTask call assignment.

Reply:
I originally had it as direct calls and I do not remember why I changed it. I think I  did it because it is easier to read. I also do not remember if I tried using `u8` or `u16`.